src/cpp/session/include/session/SessionConsoleProcessInfo.hpp

```diff
@@ -53,6 +53,7 @@ enum AutoCloseMode
    DefaultAutoClose = 0, // obey user preference
```

Comment:
Aren't you also obeying the user preference when AlwaysAutoClose or NeverAutoClose is set?

Reply:
This pattern is a bit weird, to be sure. Not entirely sure what I was thinking back in the day, but don't want to mess with it too much. There's this "default" setting that gets associated with the ConsoleProcess metadata, then when a process exits, it looks at the user preference and stamps the actual value onto it. So from then on that particular terminal will have that exact setting even if the user preference is changed again.

Reply:
So a terminal that's already opened won't be affected by a preference update?